homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -1,10 +1,12 @@
 """Support for Fritzbox binary sensors."""
 import requests
 
-from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.binary_sensor import DOMAIN, BinarySensorDevice
 
 from .const import DOMAIN as FRITZBOX_DOMAIN, LOGGER
 
+entities = set()
```

Comment:
Move this inside `async_setup_entry` and if needed store it in hass.data.

Reply:
But isn't the `async_setup_entry` called maybe multiple times when one device gets configured later? Then `entities` would be again empty and the entity gets set up a second time.

Reply:
No, `async_setup_entry` is only called one time per entry and platform.

Reply:
Yes, but all known devices are in `hass.data` and are iterated on every `async_setup_entry`. So I need an entities list which is global over all calls.

Reply:
Then set a default in hass.data. Use `dict.setdefault`.